Wikipedia/Code/MWKImage.m

```diff
@@ -278,4 +278,58 @@ - (BOOL)isLeadImage {
     return [self.article.image isEqualToImage:self];
 }
 
++ (CGSize)minimumImageSizeForGalleryInclusion {
+    return CGSizeMake(80, 80);
+}
+
+- (BOOL)isCanonical {
+    return [MWKImage fileSizePrefix:self.sourceURLString] == NSNotFound;
+}
+
+- (MWKImage *)largestNonCanonicalVariant {
+    NSString *largestNonCanonicalVariantURL = [[[self.article.images imageSizeVariants:self.sourceURLString] bk_select:^BOOL (NSString* url) {
+        return [MWKImage fileSizePrefix:url] != NSNotFound;
+    }] lastObject];
+    return [self.article imageWithURL:largestNonCanonicalVariantURL];
+}
+
+-(UIImage*)imageFromAppImageCache {
```

Comment:
this sets a bad precedent of breaking the `WMFImageController` abstraction, which for all other things intentional encapsulates the cache. instead of querying the `MWKImage` for its data, ask the `WMFImageController` for the image from its cache. for example: `WMFImageController.sharedInstance().cachedImageWithURL(self.sourceURLString)`.  having this as an instance method on `MWKImage` is acceptable IMO as a convenience, but it should still query the image controller, not the cache itself.